plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergBucketing.java

```diff
@@ -0,0 +1,174 @@
+        assertBucketEquals("int", Integer.MIN_VALUE);
+        assertBucketEquals("int", Integer.MAX_VALUE);
+
+        assertBucketEquals("long", null);
+        assertBucketEquals("long", 300_000_000_000L);
+        assertBucketEquals("long", Long.MIN_VALUE);
+        assertBucketEquals("long", Long.MAX_VALUE);
+
+        assertBucketEquals("string", null);
+        assertBucketEquals("string", "");
+        assertBucketEquals("string", "test string");
+        assertBucketEquals("string", "\u5f3a\u5927\u7684Trino\u5f15\u64ce"); // 3-byte UTF-8 sequences (in Basic Plane, i.e. Plane 0)
+        assertBucketEquals("string", "\uD843\uDFFC\uD843\uDFFD\uD843\uDFFE\uD843\uDFFF"); // 4 code points: 20FFC - 20FFF. 4-byte UTF-8 sequences in Supplementary Plane 2
+
+        assertBucketEquals("decimal(1, 1)", null);
+        assertBucketEquals("decimal(1, 1)", new BigDecimal(1));
+        assertBucketEquals("decimal(1, 1)", new BigDecimal(1));
+        assertBucketEquals("decimal(18, 10)", new BigDecimal(0));
+        assertBucketEquals("decimal(38, 10)", new BigDecimal("999999.9999999999"));
+        assertBucketEquals("decimal(38, 0)", new BigDecimal("99999999999999999999999999999999999999"));
+        assertBucketEquals("decimal(38, 10)", new BigDecimal("9999999999999999999999999999.9999999999"));
+
+        assertBucketEquals("binary", null);
+        assertBucketEquals("binary", ByteBuffer.wrap(new byte[]{}));
+        assertBucketEquals("binary", ByteBuffer.wrap("hello trino".getBytes(StandardCharsets.UTF_8)));
+        assertBucketEquals("binary", ByteBuffer.wrap("\uD843\uDFFC\uD843\uDFFD\uD843\uDFFE\uD843\uDFFF".getBytes(StandardCharsets.UTF_16)));
+    }
+
+    @Test(dataProvider = "unsupportedBucketingTypes")
+    public void testUnsupportedTypes(Type type)
+    {
+        assertThatThrownBy(() -> computeIcebergBucket(type, null, 1))
+                .hasMessage(format("Cannot bucket by type: %s", type));
+
+        assertThatThrownBy(() -> computeTrinoBucket(type, null, 1))
+                .hasMessage(format("Unsupported type for 'bucket': %s", toTrinoType(type, TYPE_MANAGER)));
+    }
+
+    @Test(dataProvider = "unsupportedTrinoBucketingTypes")
+    public void testUnsupportedTrinoTypes(Type type)
+    {
+        assertThatThrownBy(() -> computeTrinoBucket(type, null, 1))
+                .hasMessage(format("Cannot convert from Iceberg type '%s' (%s) to Trino type", type, type.typeId()));
+
+        assertNull(computeIcebergBucket(type, null, 1));
+    }
+
+    @DataProvider
+    public Object[][] unsupportedBucketingTypes()
+    {
+        return new Object[][] {
+            {BooleanType.get()},
+            {FloatType.get()},
+            {DoubleType.get()}};
+    }
+
+    @DataProvider
+    public Object[][] unsupportedTrinoBucketingTypes()
+    {
+        return new Object[][] {
+                {UUIDType.get()}};
+    }
+
+    private void assertBucketEquals(String icebergTypeName, Object value)
```

Comment:
we want to ensure two things

- bucketing is consistent between Trino and Icberg (achieved)
- bucketing is consistent across versions -- for the latter, we need to pin expected result, for example bucket number for some fixed, high bucket count (eg highest prime number within integer, not to simply chop off highest bit; or with two different bucket numbers)
   - see `expectedBuckets...` in hive bucketing tests

Reply:
sure, makes sense, I'll add assertions for that (i.e. for case Integer.MAX_VALUE...)